File: archived/src/go/ch01/1.2/l1/efficiency_demo.go
package main

import "fmt"

// For power-of-two modulus m = 2^k, n % m == n & (m-1)
func runEfficiencyDemo() {
	for _, n := range []int{5, 12, 20} {
		fmt.Printf("%d %% 8 = %d, bitmask = %d\n", n, n%8, n&7)
	}
}

File: archived/src/erlang/ch01/1.2/l1/src/efficiency_demo.erl
-module(efficiency_demo).
-export([main/0]).

%% For power-of-two modulus m = 2^k, n rem m == n band (m-1)
main() ->
    lists:foreach(
      fun(N) ->
          io:format("~p % 8 = ~p, bitmask = ~p~n", [N, N rem 8, N band 7])
      end,
      [5, 12, 20]).
